lib/vidfeeder/source_importer/youtube_playlist_importer.ex

```diff
@@ -0,0 +1,162 @@
+      {youtube_playlist_items_ids, youtube_playlist_items_map} = build_cache(youtube_playlist_items, :playlist_item_id)
+      {playlist_items_ids, playlist_items_map} = build_cache(playlist_items, :id)
+
+      %__MODULE__{
+        youtube_playlist_items: youtube_playlist_items,
+        youtube_playlist_items_ids: youtube_playlist_items_ids,
+        youtube_playlist_items_map: youtube_playlist_items_map,
+        playlist_items: playlist_items,
+        playlist_items_ids: playlist_items_ids,
+        playlist_items_map: playlist_items_map
+      }
+    end
+
+    defp build_cache(collection, key) do
+      ids = collection |> Enum.map(&Map.get(&1, key)) |> MapSet.new
+
+      map = Enum.reduce(collection, %{}, fn item, acc ->
+        key = Map.get(item, key)
+
+        Map.put(acc, key, item)
+      end)
+
+      {ids, map}
+    end
+
+    def new_playlist_items(differ) do
+      differ.playlist_items_ids
+      |> MapSet.difference(differ.youtube_playlist_items_ids)
+      |> Enum.map(fn id -> Map.get(differ.playlist_items_map, id) end)
+    end
+
+    def existing_playlist_items(differ) do
+      differ.playlist_items_ids
+      |> MapSet.intersection(differ.youtube_playlist_items_ids)
+      |> Enum.map(fn id ->
+        playlist_item = Map.get(differ.playlist_items_map, id)
+        youtube_playlist_item = Map.get(differ.youtube_playlist_items_map, id)
+
+        {youtube_playlist_item, playlist_item}
+      end)
+    end
+  end
+
+  def run(youtube_playlist) do
+    conn = YouTube.Connection.new
+
+    case YouTube.Playlist.info(conn, youtube_playlist.playlist_id) do
+      nil ->
+        Logger.debug("YouTube playlist cannot be found: #{youtube_playlist.playlist_id}")
+      playlist ->
+        IO.inspect(playlist)
+
+        if youtube_playlist.etag != playlist.etag do
+          Logger.debug("Etag mismatch")
+
+          with {:ok, youtube_playlist} <- update_playlist(youtube_playlist, playlist),
+               {:ok, youtube_playlist} <- update_playlist_items(conn, youtube_playlist) do
+            youtube_playlist
+          end
+
+        else
+          Logger.debug("Etag is the same, won't parse further")
+
+          youtube_playlist
+        end
+    end
+  end
+
+  def update_playlist(youtube_playlist, playlist) do
+    youtube_playlist
+    |> YouTubePlaylist.api_changeset(playlist)
+    |> Repo.update
+  end
+
+  def update_playlist_items(conn, youtube_playlist) do
+    playlist_items = YouTube.Playlist.items(conn, youtube_playlist.playlist_id)
+
+    youtube_videos_by_video_id =
+      conn
+      |> create_or_update_videos_from_playlist_items(playlist_items)
+      |> Enum.reduce(%{}, fn video, acc -> Map.put(acc, video.video_id, video) end)
+
+    youtube_playlist = Repo.preload(youtube_playlist, youtube_playlist_items: :youtube_video)
+    youtube_playlist_items = youtube_playlist.youtube_playlist_items
+
+    differ = YouTubePlaylistItemsDiffer.new(youtube_playlist_items, playlist_items)
```

Comment:
Change the interface to `YouTubePlaylistItemsDiffer` to simply be a `diff/2`, which returns a list of the following tuples:

```elixir
{:new, playlist_item}
{:existing, youtube_playlist_item, playlist_item}
```

Then create/update the youtube playlist item appropriately.